usecases/case_usecase.go

```diff
@@ -362,6 +367,77 @@ func (usecase *CaseUseCase) UpdateCase(
 	return updatedCase, nil
 }
 
+func (uc *CaseUseCase) Snooze(ctx context.Context, req models.CaseSnoozeRequest) error {
+	c, err := uc.repository.GetCaseById(ctx, uc.executorFactory.NewExecutor(), req.CaseId)
+	if err != nil {
+		return err
+	}
+
+	if err := uc.enforceSecurity.ReadOrUpdateCase(c, []string{c.InboxId}); err != nil {
+		return err
+	}
+
+	return uc.transactionFactory.Transaction(ctx, func(tx repositories.Transaction) error {
+		if err := uc.repository.SnoozeCase(ctx, tx, req); err != nil {
+			return err
+		}
+
+		var previousSnooze *string
+
+		if c.IsSnoozed() {
+			previousSnooze = utils.Ptr(c.SnoozedUntil.Format(time.RFC3339))
```

Comment:
@Pascal-Delange Quite unsure here, do we have another standard datetime format when we manually serialize it (without using a `timestmap`)? This seems aligned with what we return, but if it is not, I'll change it.